### PR TITLE
Export setName() for renaming folders

### LINKED ISSUE #299: Could not find function name 'setName'
I am working my way through the rcrunch documentation and am getting an error in using the `setName()` function, which is described as a quick way to change a folder name in a Crunch dataset. When I use the `setName()` function on an existing directory, however, I get a 'could not find function' error. 

screen shot 2018-07-31 at 2 58 44 pm

screen shot 2018-07-31 at 2 58 48 pm

### PR DESCRIPTION
**Pull request by jonkeane**
This resolves #299

### CODE REVIEW

File: vignettes/variable-order.Rmd [hunk 1]
```diff
@@ -159,7 +159,7 @@ ds %>%
     setName("Demographics")
 ```
 
-We can also set the names of the objects contained in a folder with `setNames()`.
+We can also set the names of the objects contained in a folder with `setName()`.
```
[the remark this answers is not in the document]
Comment (jonkeane) [reply vignettes/variable-order.Rmd]: I can add that as well, yeah.

[the remark this answers is not in the document]
Comment (nealrichardson) [reply vignettes/variable-order.Rmd]: Right, so this line change is wrong, and we need the example of setting the names of contents added below here.

**Comment by jonkeane** [comment]
Ok, `setNames()` is also available now, however, I've noticed that although the `PATCH` is accepted (with `204`), the values actually aren't being saved.

In digging further, it looks like we could use the `stats::setNames()` and get the same behavior, though the error messages aren't super friendly. I could go either way on maintaining this `setNames` just for more user-friendly error, or taking it out and relying on `stats::setNames()` to do the right thing, and have slightly less ness user-messaging. I think I'm leaning towards the second option — but could be convinced of the first.

**Comment by jonkeane** [comment]
OK, this is ready for review again. I've added wrapping for catalog `PATCH`ing and ran integration tests to make sure that those `PATCH`es are accepted by the various endpoints they go to.

**Pull request merged**